DEFWRAPPER extensions

ramenbytes opened the pull request:
First time doing a pull request, feedback welcome.

This pull request makes it possible to use :void and :pointer types in DEFWRAPPER. C-TYPE-NAME can also handle struct types like (:struct timeval). Lastly, several docstrings written while exploring the wrapper and grovel functionalities are included.

luismbo commented:
Thanks for the contribution! I haven't yet had the chance to try the code but at a first glance I'd just replace cdr and cadr with rest and second. Tests would be great! (I can help you with that if you run into issues there.)

ramenbytes commented:
I added tests and changed the functions. I also added a docstring for cffi's custom DEFTEST macro. I noticed most of the already existing tests seem to use a '.' to separate bits of information in the name. Is there any specific convention being followed with that?

ramenbytes commented:
@luismbo 
Following up on this, is there anything on my part that needs to be done for this to be merged? Is there any chance of it being merged?
